src/kernels/jupyter/jupyterConnection.ts

```diff
@@ -92,18 +93,24 @@ export class JupyterConnection implements IExtensionSyncActivationService {
     public async updateServerUri(uri: string): Promise<void> {
         const idAndHandle = this.extractJupyterServerHandleAndId(uri);
         if (idAndHandle) {
-            const serverUri = await this.jupyterPickerRegistration.getJupyterServerUri(
-                idAndHandle.id,
-                idAndHandle.handle
-            );
-            this.uriToJupyterServerUri.set(uri, serverUri);
-            // See if there's an expiration date
-            if (serverUri.expiration) {
-                const timeoutInMS = serverUri.expiration.getTime() - Date.now();
-                // Week seems long enough (in case the expiration is ridiculous)
-                if (timeoutInMS > 0 && timeoutInMS < 604800000) {
-                    this.pendingTimeouts.push(setTimeout(() => this.updateServerUri(uri).ignoreErrors(), timeoutInMS));
+            try {
+                const serverUri = await this.jupyterPickerRegistration.getJupyterServerUri(
+                    idAndHandle.id,
+                    idAndHandle.handle
+                );
+                this.uriToJupyterServerUri.set(uri, serverUri);
+                // See if there's an expiration date
+                if (serverUri.expiration) {
+                    const timeoutInMS = serverUri.expiration.getTime() - Date.now();
+                    // Week seems long enough (in case the expiration is ridiculous)
+                    if (timeoutInMS > 0 && timeoutInMS < 604800000) {
+                        this.pendingTimeouts.push(
+                            setTimeout(() => this.updateServerUri(uri).ignoreErrors(), timeoutInMS)
+                        );
+                    }
                 }
+            } catch (ex) {
+                throw new RemoteJupyterServerUriProviderError(idAndHandle.id, idAndHandle.handle, ex);
```

Comment:
Throw a custom error so we can handle this, e.g. possible the jupyter server is down and user can now choose to ignore this connection completely.